python/ray/util/state/common.py

```diff
@@ -163,6 +164,30 @@ def __post_init__(self):
                     "Available predicates: =, !=."
                 )
 
+    def has_conflicting_filters(self) -> bool:
+        # Check the filters in the ListApiOptions conflicts. Specifically for:
```

Comment:
nit: potentially a `=` with the same value and a `!=` with the same value would also result in empty results on the same key i guess - but given this is an optimization and the doc is clear. I think it's fine.

Reply:
Good point! Added a todo in the PR for the optimization